app/tests/unit/config/test_config_data_usage.py

```diff
@@ -0,0 +1,298 @@
+from src.config import Config
+
+import json
+from unittest import mock
+
+def load_config(config_name, **data):
+    from mako.template import Template
+
+    with open('./tests/unit/config/{}.json'.format(config_name)) as fd:
+        template_text = fd.read()
+
+    config_text = Template(template_text).render(**data)
+    config = json.loads(config_text)
+    return config
+
+def test_highest_config_version_chosen():
+    with mock.patch('src.config.Config.get_configs_from_dynamodb') as mock_dynamo:
+        mock_dynamo.return_value = [
+            load_config(
+                config_name = 'config_template',
+                access_map  = json.dumps({}),
+                version     = json.dumps('ver_1'),
+                uri_map     = json.dumps({})
+            ),
+            load_config(
+                config_name = 'config_template',
+                access_map  = json.dumps({}),
+                version     = json.dumps('ver_10'),
+                uri_map     = json.dumps({})
+            ),
+            load_config(
+                config_name = 'config_template',
+                access_map  = json.dumps({}),
+                version     = json.dumps('ver_2'),
+                uri_map     = json.dumps({})
+            )
+        ]
+
+        config = Config(
+            region_name='some-aws-region',
+            table_name='some-dynamodb-table',
+        )
+
+        assert config.get_version() == 'ver_10'
+
+def test_uri_map_missing_pretty_uri():
+    with mock.patch('src.config.Config.get_configs_from_dynamodb') as mock_dynamo:
+        mock_dynamo.return_value = [
+            load_config(
+                config_name = 'config_template',
+                access_map  = json.dumps({}),
+                version     = json.dumps('ver_42'),
+                uri_map     = json.dumps({
+                    '/some-pretty-path/pretty-resource': {
+                        'S': '/some-ugly-path/ugly-resource'
```

Comment:
Think calls to load config can be dry’ed up? Eg make ‘config_template’ and {} be defaults and take dictionaries and have load config call json.dumps?

Reply:
Done.